### PR TITLE
cgroups: fix the issue of get wrong online cpus

### PR DESCRIPTION
**Pull request by lifupan**
It's better to get the online cpusets from
"/sys/devices/system/cpu/online" instead of from
cpuset cgroup, cause there would be an latency
between one cpu online and present in the root
cpuset cgroup.

Fixes: #1536

Signed-off-by: fupan.lfp <fupan.lfp@antfin.com>

### CODE REVIEW

**Review by liubin — APPROVED** [review]

**Review by egernst — COMMENTED** [review]
Interesting that this became an issue!

We seem to be using the term CPUSET throughout this patch, but really what we're looking at is just the guest CPUs, right? I understand the goal is to get the CPUs so we can, if needed, update the cpuset, right?

Perhaps at least we should make clear that we're reading online CPUs, not cpusets?

File: src/agent/rustjail/src/cgroups/fs/mod.rs [hunk 1]
```diff
@@ -37,6 +37,8 @@ use std::collections::HashMap;
 use std::fs;
 use std::path::Path;
 
+const GUEST_CPUSETS_PATH: &str = "/sys/devices/system/cpu/online";
```
Comment (egernst) [inline src/agent/rustjail/src/cgroups/fs/mod.rs:40]: Maybe:

```suggestion
const GUEST_CPUS_PATH: &str = "/sys/devices/system/cpu/online";
```

**Review by fidencio — APPROVED** [review]
lgtm, thanks @lifupan!

by the way, I agree with Eric's comment, but I also think that could come with a later fix, mainly considering this one may be ready to merge (wrt to the CI).

File: src/agent/rustjail/src/cgroups/fs/mod.rs (hunk 1 above)
Comment (lifupan) [reply src/agent/rustjail/src/cgroups/fs/mod.rs:40]: @egernst Yes, you are right, and updated.

**Review by egernst — APPROVED** [review]

**Pull request merged**